---
name: UIAutomationWithError
description: This test is to check if teardown is called even when an exception is thrown in execution steps.

testScenarios:
  - name: UIScenario1
    executionSteps:
      - step: When the UI button "1" is pressed
      - step: throw a null pointer exception
      - step: And the UI button "+" is pressed
      - step: And the UI button "2" is pressed
      - step: And the UI button "=" is pressed
      - step: And the calculator should display "3"
    tearDownSteps:
      - step: continue with teardown even on exception

scenarioTearDownSteps:
  - step: continue with teardown even on exception

tearDownSteps:
  - step: continue with teardown even on exception
